packages/dds/merge-tree/src/test/client.rollbackFarm.spec.ts

```diff
@@ -0,0 +1,279 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { strict as assert } from "assert";
+import * as fs from "fs";
+import random from "random-js";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { SegmentGroup } from "../mergeTreeNodes";
+import { IMergeTreeOp, MergeTreeDeltaType } from "../ops";
+import {
+    annotateRange,
+    applyMessages,
+    doOverRange,
+    IConfigRange,
+    IMergeTreeOperationRunnerConfig,
+    insertAtRefPos,
+    removeRange,
+    ReplayGroup,
+    replayResultsPath,
+    TestOperation,
+    generateClientNames,
+} from "./mergeTreeOperationRunner";
+import { TestClient } from "./testClient";
+import { TestClientLogger } from "./testClientLogger";
+
+interface IRollbackFarmConfig extends IMergeTreeOperationRunnerConfig {
+    minLength: IConfigRange;
+    rollbackOnlyClients: IConfigRange;
+    editOnlyClients: IConfigRange;
+    rollbackAndEditClients: IConfigRange;
+    opsPerRollbackRange: IConfigRange;
+}
+
+interface IClients {
+    clients: TestClient[];
+    rollbackOnlyCount: number;
+    editOnlyCount: number;
+    rollbackAndEditCount: number;
+}
+
+const allOperations: TestOperation[] = [
+    removeRange,
+    annotateRange,
+    insertAtRefPos,
+];
+
+export const debugOptions: IRollbackFarmConfig = {
+    minLength: { min: 2, max: 2 },
+    rollbackOnlyClients: { min: 0, max: 0 },
+    editOnlyClients: { min: 0, max: 0 },
+    rollbackAndEditClients: { min: 1, max: 1 },
+    opsPerRollbackRange: { min: 1, max: 15 },
+    opsPerRoundRange: { min: 1, max: 100 },
+    rounds: 5,
+    operations: allOperations,
+    incrementalLog: true,
+    growthFunc: (input: number) => input + 1,
+};
+
+export const defaultOptions: IRollbackFarmConfig = {
+    minLength: { min: 1, max: 512 },
+    rollbackOnlyClients: { min: 1, max: 1 },
+    editOnlyClients: { min: 1, max: 1 },
+    rollbackAndEditClients: { min: 1, max: 3 },
+    opsPerRollbackRange: { min: 1, max: 128 },
+    opsPerRoundRange: { min: 1, max: 128 },
```

Comment:
i would probably just go with a fixed number of ops, like 10, for non-rollback, this times rounds will represent the ops on any merge tree, so will still get lots of coverage